added malayalam language

shyjun opened the pull request:
added malayalam language

ShyjuNarangaprath committed:
added malayalam language. append en-utf8.csv

sspanak commented on app/languages/dictionaries/ml-utf8.csv:
```diff
@@ -0,0 +1,831 @@
+aa
```
A list of 800 words isn't going to be usable. Such short lists are suitable for full-sized QWERTY-like keyboards, where you can type all words even without suggestions from the keyboard. On the contrary, dictionaries for T9 keyboards must contain thousands or even millions of words to produce meaningful results. It's because the keyboard must "guess" what words match the digits.

sspanak reviewed (changes requested):
I can't accept this pull request as-is. The resulting language will not be usable at all. See my previous comments.

sspanak commented:
> Yes, i understand that it can't be merged. Thats why kept it as Draft.
> 
> The Malayalam keyboard/words I am referring is like an extension to the English words.
> 
> Probably, what I have to do is, take English word list, and add Malayalam words with it and publish it as Manglish keyboard.

It sounds like a good idea. You may choose to use a British English or an American English, or even an Indian English word base depending on what's more appropriate. Still, just 800 Malayalam words sound very little. Shouldn't there be at least tens of thousands of words? Comparing to the Hinglish dictionary in TT9, I can see maybe half of the words are in Hindi and half are in English.

Perhaps, you can make use of some word lists from [the Indic Project](https://gitlab.com/indicproject/dictionaries)? Or, if there is an official Malayalam dictionary, maybe use that instead? Of course, the words would need to be converted to Latin but it shouldn't be a problem, I believe.

shyjun commented:
> > Yes, i understand that it can't be merged. Thats why kept it as Draft.
> > The Malayalam keyboard/words I am referring is like an extension to the English words.
> > Probably, what I have to do is, take English word list, and add Malayalam words with it and publish it as Manglish keyboard.
> 
> It sounds like a good idea. You may choose to use a British English or an American English, or even an Indian English word base depending on what's more appropriate. Still, just 800 Malayalam words sound very little. Shouldn't there be at least tens of thousands of words? Comparing to the Hinglish dictionary in TT9, I can see maybe half of the words are in Hindi and half are in English.
> 
> Perhaps, you can make use of some word lists from [the Indic Project](https://gitlab.com/indicproject/dictionaries)? Or, if there is an official Malayalam dictionary, maybe use that instead? Of course, the words would need to be converted to Latin but it shouldn't be a problem, I believe.

Yes, there are thousands of words in Malayalam, but these are the words that I use regularly. The word list is maintained in https://github.com/shyjun/manglish_word_list/blob/main/words.txt

In fact, I created this word list using the export feature of TT9 and TouchPal keyboard.

Going forward, while typing, when i type a Malayalam word which is not in user words list, I will immediately add it to user words., and occationaly(once in 6 months or so) i will export it to cvs file, and update https://github.com/shyjun/manglish_word_list/blob/main/words.txt with the new words.  Probably once in a year or so, i will create PR with dictionary update

Hope you are ok with this approach.

shyjun commented:
thanks for the comments. as you suggested, i took words from the Indic Project(https://gitlab.com/indicproject/dictionaries), and added my regularly used words and updated the MR.

please confirm if this approach is ok, if so, i will make further changes, and mark the PR as ready.

Thanks

sspanak commented on app/languages/definitions/Malayalam.yml:
```diff
@@ -0,0 +1,14 @@
+locale: ml
+dictionaryFile: ml-utf8.csv
+abcString: xyz
```
This is not truly Malayalam but rather a mixture of English and Malayalam. It should have a different locale and name, something like Hinglish, which is a mixture of Hindi and English.

Also, the locale must consist of four letters, not two. English is the only exception.

Let's do the following:
```suggestion
locale: ml-GB
name: Malayalam / Latin
dictionaryFile: malayalam-latin-utf8.csv
```

Let's as well rename the `.yml` file to `MalayalamLatin.yml` for consistency with Tamazight Latin.

Also, must add icons.

sspanak commented on docs/dictionaries/mlWordlistReadme.txt:
```diff
@@ -0,0 +1,5 @@
+Malayalam word list by: Shyju N
```
This file should also include a link to the Indic project.

shyjun commented:
My plan is to create 2 dictionaries. First is for Malayalam, and second is Manglish(Malayalam + English).

The word Manglish is commonly used which means writing Malayalam in English letters.

Malayalam language will have only Malayalam words, and Manglish will have Malayalam + English words.

Let me know if this approach is ok, if so I will update the PR
